crates/ruff_python_ast/src/nodes.rs

```diff
@@ -1351,6 +1353,72 @@ impl Ranged for FStringPart {
     }
 }
 
+pub trait AbstractStringFlags: Copy {
+    /// Does the string use single or double quotes in its opener and closer?
+    fn quote_style(self) -> Quote;
+
+    /// Is the string triple-quoted, i.e.,
+    /// does it begin and end with three consecutive quote characters?
+    fn is_triple_quoted(self) -> bool;
+
+    fn is_raw_string(self) -> bool;
+
+    fn prefix(self) -> impl AbstractStringPrefix;
```

Comment:
Yeah, makes sense. `AnyStringPrefix` has all of the same methods, and is strictly more precise, since you can narrow it later to figure out which variant you have.